lib/services/local/mfa.go

```diff
@@ -0,0 +1,172 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+package local
+
+import (
+	"bytes"
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb" //nolint:depguard // needed because mfav1.ValidatedMFAChallenge uses gogoproto
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	mfav1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/mfa/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local/generic"
+)
+
+// MFAService implements the storage layer for MFA resources.
+type MFAService struct {
+	logger *slog.Logger
+	svc    *generic.ServiceWrapper[*validatedMFAChallenge]
+}
+
+// NewMFAService returns a new MFA storage service.
+func NewMFAService(b backend.Backend) (*MFAService, error) {
+	svc, err := generic.NewServiceWrapper(
+		generic.ServiceConfig[*validatedMFAChallenge]{
+			Backend:       b,
+			ResourceKind:  types.KindValidatedMFAChallenge,
+			BackendPrefix: backend.NewKey(types.KindValidatedMFAChallenge),
+			MarshalFunc:   MarshalValidatedMFAChallenge,
+			UnmarshalFunc: UnmarshalValidatedMFAChallenge,
+			ValidateFunc:  validateValidatedMFAChallenge,
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &MFAService{
+		logger: slog.With(teleport.ComponentKey, "MFAService.local"),
```

Comment:
I noticed we are not really consistent in this package with these names. Are we good with this or do we have better name in mind?

Reply:
I would ditch the ".local", unless you actually plan on having distinct implementations.

So: 

```suggestion
		logger: slog.With(teleport.ComponentKey, "MFAService"),
```

Reply:
Also note that, if you scope the RPC service in the same manner, they'd log with the same component name. This may or may not be desirable.